test/StringProcessing/Parse/forward-slash-regex.swift

```diff
@@ -309,17 +308,84 @@ quxqux(!/^/)
 // expected-error@-2 {{cannot convert value of type 'Regex<Substring>' to expected argument type 'Bool'}}
 
 quxqux(/^)
-
-do {
-  quxqux(/^) / 1
-  // expected-error@-1 {{closing ')' does not balance any groups openings}}
-  // expected-error@-2 {{expected ',' separator}}
-}
+_ = quxqux(/^) / 1
 
 let arr: [Double] = [2, 3, 4]
 _ = arr.reduce(1, /) / 3
 _ = arr.reduce(1, /) + arr.reduce(1, /)
 
+// MARK: ')' disambiguation behavior
+
+_ = (/x)
+// expected-error@-1 {{'/' is not a prefix unary operator}}
+
+_ = (/x)/
+// expected-error@-1 {{'/' is not a prefix unary operator}}
+// expected-error@-2 {{'/' is not a postfix unary operator}}
+
+_ = (/[(0)])/
+// expected-error@-1 {{'/' is not a prefix unary operator}}
+// expected-error@-2 {{'/' is not a postfix unary operator}}
+
+_ = /[(0)]/
+_ = /(x)/
+_ = /[)]/
+_ = /[a\])]/
+_ = /([)])/
+_ = /]]][)]/
+
+_ = /
+// expected-error@-1 {{unterminated regex literal}}
+
+_ = /)
+// expected-error@-1 {{unterminated regex literal}}
+// expected-error@-2 {{closing ')' does not balance any groups openings}}
+
+let fn: (Int, Int) -> Int = (/)
+
+_ = /\()/
+// expected-error@-1 {{'/' is not a prefix unary operator}}
+// expected-error@-2 {{'/' is not a postfix unary operator}}
+// expected-error@-3 {{invalid component of Swift key path}}
+
```

Comment:
Could you add tests for `\)` being ignored? e.g. `_ = (/whatever\)/` should be a regex inside a paren expression with missing `)`

Reply:
There is a test for `/\)/`, but added your example too, thanks!